lib/spack/spack/stage.py

```diff
@@ -455,30 +433,18 @@ def restage(self):
         self.fetcher.reset()
 
     def create(self):
-        """Creates the stage directory.
-
-        If get_tmp_root() is None, the stage directory is created
-        directly under spack.paths.stage_path, otherwise this will attempt to
-        create a stage in a temporary directory and link it into
-        spack.paths.stage_path.
-
         """
-        # Create the top-level stage directory
-        mkdirp(spack.paths.stage_path)
-        remove_if_dead_link(self.path)
-
-        # If a tmp_root exists then create a directory there and then link it
-        # in the stage area, otherwise create the stage directory in self.path
-        if self._need_to_create_path():
-            tmp_root = get_tmp_root()
-            if tmp_root is not None:
-                # tempfile.mkdtemp already sets mode 0700
-                tmp_dir = tempfile.mkdtemp('', _stage_prefix, tmp_root)
-                tty.debug('link %s -> %s' % (self.path, tmp_dir))
-                os.symlink(tmp_dir, self.path)
-            else:
-                # emulate file permissions for tempfile.mkdtemp
-                mkdirp(self.path, mode=stat.S_IRWXU)
+        Ensures the top-level (config:build_stage) directory exists.
+        """
+        # Emulate file permissions for tempfile.mkdtemp.
+        if not os.path.exists(self.path):
+            print("TLD: %s does not exist, creating it" % self.path)
```

Comment:
It looks like you forgot to remove these print statements. Also, I'm not sure how I feel about the logic here. If I install a package where several of the dependencies are already installed, I see:
```
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/py-pyqt5-5.13.0-q3dc5mhfrsqg2xzdlwkfjmmszi6vkj7n does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/bzip2-1.0.8-llnc6sqphf7yp4drig6njzzkxctjwfse does not exist, creating it
==> bzip2 is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/bzip2-1.0.8-llnc6sqphf7yp4drig6njzzkxctjwfse
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/expat-2.2.5-ivx3l77irr7xnft6v2xduzqier7zlct3 does not exist, creating it
==> expat is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/expat-2.2.5-ivx3l77irr7xnft6v2xduzqier7zlct3
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/ncurses-6.1-nhodl3ckldkmn6nfzgtymzlsn7mm3htq does not exist, creating it
==> ncurses is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/ncurses-6.1-nhodl3ckldkmn6nfzgtymzlsn7mm3htq
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/readline-7.0-kn3u3qhqwktlvjpr47ism2uqqgstvvf2 does not exist, creating it
==> readline is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/readline-7.0-kn3u3qhqwktlvjpr47ism2uqqgstvvf2
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/gdbm-1.18.1-pvjwideh43qybd2ifi65gjxq2dd6p4nj does not exist, creating it
==> gdbm is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/gdbm-1.18.1-pvjwideh43qybd2ifi65gjxq2dd6p4nj
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libffi-3.2.1-rmioaofogfk7efvt3cpb3d433sevjnaf does not exist, creating it
==> libffi is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libffi-3.2.1-rmioaofogfk7efvt3cpb3d433sevjnaf
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/zlib-1.2.11-nqnlnxddpa4qfz5cnf4grnct57ds25ks does not exist, creating it
==> zlib is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/zlib-1.2.11-nqnlnxddpa4qfz5cnf4grnct57ds25ks
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/openssl-1.1.1b-z6erpibfcvzxo5chnvovu3qmet4dzuuv does not exist, creating it
==> openssl is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/openssl-1.1.1b-z6erpibfcvzxo5chnvovu3qmet4dzuuv
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/sqlite-3.28.0-mijtevf42dqzctgyaizz44skbyqwxgol does not exist, creating it
==> sqlite is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/sqlite-3.28.0-mijtevf42dqzctgyaizz44skbyqwxgol
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/xz-5.2.4-snwzcouby3x3ztkutupqzmqwbbxh2tur does not exist, creating it
==> xz is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/xz-5.2.4-snwzcouby3x3ztkutupqzmqwbbxh2tur
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/python-3.7.4-jqlxzxpwzdhmrirodxanluofnce5o3ab does not exist, creating it
==> python is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/python-3.7.4-jqlxzxpwzdhmrirodxanluofnce5o3ab
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/py-sip-4.19.18-mqzm6offewfpn4amozlz2gdcntal4jvq does not exist, creating it
==> py-sip is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/py-sip-4.19.18-mqzm6offewfpn4amozlz2gdcntal4jvq
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/double-conversion-2.0.1-mf7korxtnck5wcpydk4bwushpzubp7vy does not exist, creating it
==> double-conversion is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/double-conversion-2.0.1-mf7korxtnck5wcpydk4bwushpzubp7vy
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libpng-1.6.34-h24accpckijen4jyzgawyxyoyvqgeb4i does not exist, creating it
==> libpng is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libpng-1.6.34-h24accpckijen4jyzgawyxyoyvqgeb4i
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/freetype-2.9.1-l4ys7nhdhzpv7bgvzh2frzdofkbg3j4l does not exist, creating it
==> freetype is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/freetype-2.9.1-l4ys7nhdhzpv7bgvzh2frzdofkbg3j4l
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libiconv-1.15-6y7tjlqeuu7qmpv3zokgbdowqs7emk7y does not exist, creating it
==> libiconv is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libiconv-1.15-6y7tjlqeuu7qmpv3zokgbdowqs7emk7y
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libxml2-2.9.9-owmrtw347mtch4jm4cium25nvsjamqkf does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-xmlts-owmrtw347mtch4jm4cium25nvsjamqkf does not exist, creating it
==> libxml2 is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libxml2-2.9.9-owmrtw347mtch4jm4cium25nvsjamqkf
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/tar-1.31-vyjf7fdqrr6ybqblhuujonbcguepjbne does not exist, creating it
==> tar is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/tar-1.31-vyjf7fdqrr6ybqblhuujonbcguepjbne
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/gettext-0.19.8.1-n3673flplqiagvozwgyeezzs4u22gvrr does not exist, creating it
==> gettext is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/gettext-0.19.8.1-n3673flplqiagvozwgyeezzs4u22gvrr
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/pcre-8.42-mgfx4tewjjhhbggofpwqgbto2474uuuk does not exist, creating it
==> pcre is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/pcre-8.42-mgfx4tewjjhhbggofpwqgbto2474uuuk
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/perl-5.26.2-agbs3wmvu2qp3hjc3amdhpo4znwkcwxs does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-cpanm-agbs3wmvu2qp3hjc3amdhpo4znwkcwxs does not exist, creating it
==> perl is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/perl-5.26.2-agbs3wmvu2qp3hjc3amdhpo4znwkcwxs
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/glib-2.56.3-32dbs5w24drcs5rioahldtsfaafxujxt does not exist, creating it
==> glib is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/glib-2.56.3-32dbs5w24drcs5rioahldtsfaafxujxt
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/font-util-1.3.1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-encodings-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-alias-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-ttf-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-75dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-type1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-dec-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-jis-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-sun-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-100dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-ibm-type1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-isas-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-misc-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-mutt-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-micro-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-adobe-75dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-arabic-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-cursor-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-daewoo-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-misc-meltho-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-adobe-100dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-misc-cyrillic-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-misc-ethiopic-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-xfree86-type1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bitstream-75dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bitstream-type1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-cronyx-cyrillic-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-schumacher-misc-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-screen-cyrillic-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bitstream-100dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bitstream-speedo-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-winitzki-cyrillic-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-adobe-utopia-75dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-adobe-utopia-type1-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-adobe-utopia-100dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-lucidatypewriter-75dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/resource-font-bh-lucidatypewriter-100dpi-jkc3lwybt5drczabq3wsxlhovcjhc6oy does not exist, creating it
==> font-util is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/font-util-1.3.1-jkc3lwybt5drczabq3wsxlhovcjhc6oy
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/fontconfig-2.12.3-ktmcn2j2yuyrgu5fz67zubcsutyrjc44 does not exist, creating it
==> fontconfig is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/fontconfig-2.12.3-ktmcn2j2yuyrgu5fz67zubcsutyrjc44
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/pixman-0.38.0-uyvgo633ewb3bqlnpduyl2dviquhlhxk does not exist, creating it
==> pixman is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/pixman-0.38.0-uyvgo633ewb3bqlnpduyl2dviquhlhxk
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/cairo-1.16.0-vgqs3ecslekb2kmcntes7hg6sphap4ga does not exist, creating it
==> cairo is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/cairo-1.16.0-vgqs3ecslekb2kmcntes7hg6sphap4ga
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/icu4c-60.1-iuyrtmmlyd6dlqs2rxsxuy44v3tfnaku does not exist, creating it
==> icu4c is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/icu4c-60.1-iuyrtmmlyd6dlqs2rxsxuy44v3tfnaku
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/harfbuzz-2.3.1-uee73fo27ad66tjhaejukg6qrz3k5cpr does not exist, creating it
==> harfbuzz is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/harfbuzz-2.3.1-uee73fo27ad66tjhaejukg6qrz3k5cpr
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libjpeg-turbo-2.0.2-hzhdmt6d73ina3jzdzxh6n44kvgmyg6o does not exist, creating it
==> libjpeg-turbo is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libjpeg-turbo-2.0.2-hzhdmt6d73ina3jzdzxh6n44kvgmyg6o
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libtiff-4.0.10-7skodcsbdqpfjtxmsp26t5hzgpzvfvfd does not exist, creating it
==> libtiff is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libtiff-4.0.10-7skodcsbdqpfjtxmsp26t5hzgpzvfvfd
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/lcms-2.9-faanb5fcylzjueoepefp37tkyt75svay does not exist, creating it
==> lcms is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/lcms-2.9-faanb5fcylzjueoepefp37tkyt75svay
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/libmng-2.0.3-lz2tuhauvupyf2aeka2uqrdjvyog2qng does not exist, creating it
==> libmng is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/libmng-2.0.3-lz2tuhauvupyf2aeka2uqrdjvyog2qng
==> opengl@2.1 : externally installed in /opt/X11
==> opengl@2.1 : already registered in DB
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/pcre2-10.31-sbr6r5l4ghlt3rabeawheyrih3hpme2y does not exist, creating it
==> pcre2 is already installed in /Users/Adam/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.1-apple/pcre2-10.31-sbr6r5l4ghlt3rabeawheyrih3hpme2y
TLD: /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/spack-stage/Adam/qt-5.11.3-xuifxuxq6rxho2sswa7vpfqhdmq5epif does not exist, creating it
```
We should not be creating stages for packages that are already installed.

Reply:
My apologies.  The print statements are removed in #12474 .